data/lib/src/datasource/authentication_datasource.dart

```diff
@@ -65,6 +69,9 @@ class AuthenticationDataSource {
     });
   }
 
+  bool isAuthenticateWithOTPError(String authErrorCode) =>
+      authErrorCode != null && (authErrorCode == '1002' || authErrorCode == '1003');
```

Comment:
I think it's a good idea to create an error list for easy testing and expansion later. `['1002', '1003']`

Reply:
hmm, error code list for only OTP case, I don't think we need that. If we have errorCode for other cases than OTP error then we have to use another list. this is not generic enough IMO.

Reply:
it is so magical, please use constant instead

Reply:
done